pennylane/hf/tapering.py

```diff
@@ -464,3 +467,106 @@ def transform_hamiltonian(h, generators, paulix_ops, paulix_sector):
     c = qml.math.stack(c)
 
     return _simplify(qml.Hamiltonian(c, o))
+
+
+def _sector_energy(sector):
+    """Computes energy of the given tapered Hamiltonian.
+
+    Args:
+        sector (pennylane.Hamiltonian): tapered Hamiltonian whose energy has to be calculated
+
+    Returns:
+        energy (float): energy of the tapered hamiltonian
+    """
+
+    if len(sector.wires) < 2:
+        energy = min(sp.linalg.eigvals(qml.utils.sparse_hamiltonian(sector).toarray()))
+    else:
+        energy = sp.sparse.linalg.eigs(qml.utils.sparse_hamiltonian(sector), k=2)[0].min()
+
+    return energy
+
+
+def find_optimal_sector(qubit_op, generators, paulix_ops, brute_force=True, num_electrons=0):
+    r"""Get the optimal sector which contains the ground state.
+
+    To obtain the optimal sector one can brute force through all the permutation or by utilize the following
+    relation between the Pauli-Z qubit operator and the occupation number under Jordan-Wigner transform.
+
+    .. math::
+
+        \sigma_z^{i} = a_{i}^{\dagger}a_{i} - 1
+
+    This relation allows us to figure out whether the orbital is occupied or unoccupied in a given symmetry sector,
+    to build the correct eigensector.
```

Comment:
This can be explained a bit better.

Reply:
Agree. How exactly does knowing if an orbital is occupied or not help us build the correct eigensector?